tmt/steps/provision/beaker_mrack.py

```diff
@@ -0,0 +1,532 @@
+import dataclasses
+import datetime
+import sys
+from typing import Any, Dict, List, Optional, Tuple, cast
+
+import click
+
+import tmt
+import tmt.options
+import tmt.steps
+import tmt.steps.provision
+import tmt.utils
+from tmt.utils import ProvisionError, updatable_message
+
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+else:
+    from typing_extensions import TypedDict
+
+from asyncio import get_event_loop
+from asyncio.coroutines import coroutine  # type: ignore
+from functools import update_wrapper
+
+from mrack.providers import providers
+from mrack.providers.beaker import PROVISIONER_KEY as BEAKER
+from mrack.providers.beaker import BeakerProvider
+from mrack.transformers.beaker import BeakerTransformer
+
+import mrack
+
+
+def async_run(func: Any) -> Any:
+    """Decorate click actions to run as async."""
+    func = coroutine(func)
+
+    def wrapper(*args: Dict[Any, Any], **kwargs: Dict[Any, Any]) -> Any:
+        loop = get_event_loop()
+        return loop.run_until_complete(func(*args, **kwargs))
+
+    return update_wrapper(wrapper, func)
+
+
+providers.register(BEAKER, BeakerProvider)
+
+DEFAULT_USER = 'root'
+DEFAULT_ARCH = 'x86_64'
+DEFAULT_KEYNAME = 'default'
+DEFAULT_PROVISION_TIMEOUT = 3600  # 1 hour timeout at least
+DEFAULT_PROVISION_TICK = 60  # poll job each minute
+
+
+# Type annotation for "data" package describing a guest instance. Passed
+# between load() and save() calls
+GuestInspectType = TypedDict(
+    'GuestInspectType', {
+        "status": str,
+        "system": str,
+        'address': Optional[str]
+        }
+    )
+
+size_translation = {
+    "TB": 1048576,
+    "GB": 1024,
+    "MB": 1,
+    }
+
+
+@dataclasses.dataclass
+class BeakerGuestData(tmt.steps.provision.GuestSshData):
+    # Override parent class with our defaults
+    user: str = DEFAULT_USER
+
+    # Guest request properties
+    arch: str = DEFAULT_ARCH
+    image: Optional[str] = None
+    hardware: Dict[str, Any] = dataclasses.field(default_factory=dict)
+    pool: Optional[str] = None
+    keyname: str = DEFAULT_KEYNAME
+    user_data: Dict[str, str] = dataclasses.field(default_factory=dict)
```

Comment:
`pool`, `keyname` and `user_data` seem to come from the Artemis plugin (was that your inspiration?). I wonder whether they make sense for the Beaker plugin.